--- xdg-user-dirs-gtk-0.9/user-dirs-update-gtk.desktop.in.kde	2012-04-03 17:28:59.000000000 +0200
+++ xdg-user-dirs-gtk-0.9/user-dirs-update-gtk.desktop.in	2012-04-04 09:26:52.188782774 +0200
@@ -1,10 +1,8 @@
 [Desktop Entry]
-Encoding=UTF-8
 Exec=xdg-user-dirs-gtk-update
 _Name=User folders update
 _Comment=Update common folders names to match current locale
 Terminal=false
-OnlyShowIn=GNOME;LXDE;Unity;
 Type=Application
 StartupNotify=false
 X-KDE-autostart-after=panel
