diff -Nrup xdg-user-dirs-gtk-0.8/user-dirs-update-gtk.desktop.in modified//user-dirs-update-gtk.desktop.in
--- xdg-user-dirs-gtk-0.8/user-dirs-update-gtk.desktop.in	2008-09-05 21:55:20.000000000 -0300
+++ modified//user-dirs-update-gtk.desktop.in	2011-01-19 16:52:05.336363993 -0200
@@ -1,10 +1,8 @@
 [Desktop Entry]
-Encoding=UTF-8
 Exec=xdg-user-dirs-gtk-update
 _Name=User folders update
 _Comment=Update common folders names to match current locale
 Terminal=false
-OnlyShowIn=GNOME;
 Type=Application
 StartupNotify=false
 X-KDE-autostart-after=panel
